src/deadline/job_attachments/download.py

```diff
@@ -372,48 +372,20 @@ def download_files_in_directory(
     )
 
 
-def download_file(
-    file: RelativeFilePath,
-    hash_algorithm: HashAlgorithm,
-    local_download_dir: str,
+def download_file_with_s3_key(
     s3_bucket: str,
-    cas_prefix: Optional[str],
-    s3_client: Optional[BaseClient] = None,
+    s3_key: str,
+    transfer_manager,
+    local_file_name: Path,
+    file_bytes: Optional[int] = 0,
     session: Optional[boto3.Session] = None,
-    modified_time_override: Optional[float] = None,
     progress_tracker: Optional[ProgressTracker] = None,
     file_conflict_resolution: Optional[FileConflictResolution] = FileConflictResolution.CREATE_COPY,
-) -> Tuple[int, Optional[Path]]:
-    """
-    Downloads a file from the S3 bucket to the local directory. `modified_time_override` is ignored if the manifest
-    version used supports timestamps.
-    Returns a tuple of (size in bytes, filename) of the downloaded file.
-    - The file size of 0 means that this file comes from a manifest version that does not provide file sizes.
-    - The filename of None indicates that this file has been skipped or has not been downloaded.
-    """
-    if not s3_client:
-        s3_client = get_s3_client(session=session)
-
-    transfer_manager = get_s3_transfer_manager(s3_client=s3_client)
-
-    # The modified time in the manifest is in microseconds, but utime requires the time be expressed in seconds.
-    modified_time_override = file.mtime / 1000000  # type: ignore[attr-defined]
-
-    file_bytes = file.size
-
-    # Python will handle the path separator '/' correctly on every platform.
-    local_file_name = Path(local_download_dir).joinpath(file.path)
-
-    s3_key = (
-        f"{cas_prefix}/{file.hash}.{hash_algorithm.value}"
-        if cas_prefix
-        else f"{file.hash}.{hash_algorithm.value}"
-    )
-
+):
     # If the file name already exists, resolve the conflict based on the file_conflict_resolution
     if local_file_name.is_file():
         if file_conflict_resolution == FileConflictResolution.SKIP:
-            return (file_bytes, None)
+            return None
```

Comment:
Is there any change of the behavior by modifying return statement here, any tests to validate?

Reply:
I've tested with the existing tests of `download_file`, I've only refactored the return to be elsewhere, after the `download_file_with_s3_key` helper has returned; you can find the original `(file_bytes, None)` at line 485

Reply:
If there is another return None statement with `download_file_with_s3_key`, does it always makes sense for `download_file` to return `(file_bytes, None)`? Is that explicitly stated anywhere?